lr_face/models.py

```diff
@@ -34,6 +36,28 @@ def __init__(self):
         super().__init__([Input(shape=(100, 100, 3)), Flatten(), Dense(100)])
 
 
+class FaceRecognition(tf.keras.Sequential):
+    """
+    A Face Recognition model that takes RGB images with dimensions as
+    input and outputs random embeddings with dimensionality 128. This keras
+    model won't be used, but the functions will be called directly from the
+    face_recognition library. Resolution (100, 100) is not used.'
+    """
+
+    def __init__(self):
+        super().__init__([Input(shape=(100, 100, 3)), Flatten(), Dense(128)])
+
+    def predict(self, x):
+        # embed = None
+        embed = np.ones(128)
+        try:
+            embed = face_recognition.face_encodings(x)[0]
+        except IndexError:
+            # If no face found, predict returns a embeddings vector of ones.
+            print('no face found')
+        return embed
```

Comment:
I finally found the bug! This should be [embed], not embed

Downstream again the first element is taken, which resulted in embeddings of dimension 1, and crappy performance